lib/wasi/src/runtime/task_manager/mod.rs

```diff
@@ -70,6 +89,17 @@ pub trait VirtualTaskManager: std::fmt::Debug + Send + Sync + 'static {
         spawn_type: SpawnType,
     ) -> Result<(), WasiThreadError>;
 
+    /// Starts an WebAssembly task will will run on a dedicated thread
+    /// pulled from the worker pool that has a stateful thread local variable
+    /// After the trigger has successfully completed
+    fn resume_wasm_after_trigger(
```

Comment:
Why do these two need to be on the task manager?

Do we actually need custom resume logic depending on the environment?

Reply:
I know you mentioned postMessage, but I need some additional context there.

Reply:
I'll write the `wasmer-web` version and link the code here - that will be easier to understand

Reply:
i've managed to remove this now with some refactoring